web/controllers/userController/index.ts

```diff
@@ -247,10 +245,11 @@ export async function putSubjectsHandler(req: Request, res: Response) {
                     elem.maxGrade <= 13 &&
                     checkSubject(elem.name)) {
 
-                    let subject = new ApiSubject();
-                    subject.name = elem.name;
-                    subject.minGrade = elem.minGrade;
-                    subject.maxGrade = elem.maxGrade;
+                    let subject: ApiSubject = {
+                        name: elem.name,
+                        minGrade: elem.minGrade,
+                        maxGrade: elem.maxGrade
+                    };
```

Comment:
So you’re basically right. This is an issue and the code here isn’t very clean. That code really allows pupils to use the subject format (with `minGrade` and `maxGrade`) that only students should use. 
As far as I know, the frontend will always call this API endpoint with a default `minGrade` of `1` and a default `maxGrade` of `13`. So in our production database there are really a lot of subject entries with `minGrade` and `maxGrade` data. This is not good and pretty ugly, but for now doesn’t affect other parts of the code because for pupils it just ignores the additional `min`- and `maxGrade` info. 
So we should change this, but then I guess it also need to be changed in the frontend so that no errors are thrown there.

Reply:
Alright. I will open an issue for this topic.